src/main.rs

```diff
@@ -69,21 +75,76 @@ impl Group {
 struct DWebBackend {
     path: String,
     port: u16,
+    store: Store,
+    veilid_api: Option<VeilidAPI>,
 }
 
 impl DWebBackend {
-    pub fn new(path: String, port: u16) -> Self {
-        DWebBackend { path, port }
-    }
-    pub async fn start(&mut self) -> Result<()> {
-        // Init veilid
-        println!("Starting in {} with port {}", self.path, self.port);
+    pub fn new(base_path: &str, port: u16) -> Self {
+        let store_path = format!("{}/store.db", base_path);
+        let store = Store::persistent(&store_path).expect("Failed to create persistent store");
```

Comment:
Instead of throwing a panic here we should return a failed result.
```suggestion
        let store = Store::persistent(&store_path)?;
```

Reply:
The return should be a `Result<Self>` I suppose?

Reply:
In fact, it returned `DWebBackend` directly instead of a `Result<DWebBackend, E>.`
I will change the return type of the `new` method to return an appropriate error type that we can propagate.

Reply:
When trying to propagate the error using ?, the following error occurs:

`error[E0277]: the trait bound `anyhow::Error: std::error::Error` is not satisfied`

This happens because ` the error type returned by Store::persistent does not implement the std::error::Error trait, which is required for the ? operator to work seamlessly with the Result type in Rust.`

Solution implemented:

`let store = Store::persistent(&store_path).map_err(|e| anyhow!("Failed to create persistent store: {}", e))?;`

- map_err method is used to convert the error type returned by Store::persistent into an anyhow::Error type.

- The anyhow! macro is used to create a new error with a custom message and include the original error's details.

- it allows to handle the error without panicking and propagate it correctly, meeting the requirement of the ? operator.

Reply:
Is `anyhow` better than `std::error`? Feels like using the std might help avoid having to do these map_err calls all over the place. Is it what Veilid uses?

Reply:
I would not say anyhow is better but there are trade-offs in using one vs the other.

From my understanding, anyhow is good for its ease of use and to avoid boilerplate code such as error types definition.